lib/fog/image/openstack/v2/models/images.rb

```diff
@@ -17,10 +17,9 @@ def summary(options = {})
           end
 
           def find_by_id(id)
-            image_hash = service.get_image_by_id(id).body
-            Fog::Image::OpenStack::V2::Image.new(
-              image_hash.merge(:service => service)
-            )
+            result = all.find { |image| image.id == id }
```

Comment:
I can't find the comment on the outdated code anymore, so here it goes again: note the tiny difference that I'm not suggesting to initialize a new Image, but a collection:

```ruby
def find_by_id(id)
  new(service.get_image_by_id(id).body)
end
```